services/worker-manager/src/provider_google.js

```diff
@@ -22,86 +65,502 @@ class GoogleProvider extends Provider {
+          project: this.project,
+          region,
+        })).data.zones;
+      }
+      const zones = this.zonesByRegion[region];
+      const zone = zones[Math.floor(Math.random() * zones.length)].split('/').slice(-1)[0];
+
+      // This must be unique to currently existing instances and match [a-z]([-a-z0-9]*[a-z0-9])?
+      // The lost entropy from downcasing, etc should be ok due to the fact that
+      // only running instances need not be identical.
+      const instanceName = `${workerType.name}-${slugid.nice().replace(/_/g, '-').toLowerCase()}`;
+
+      let op;
+
+      try {
+        const res = await this.compute.instances.insert({
+          project: this.project,
+          zone,
+          requestId: uuid.v4(), // This is just for idempotency
+          requestBody: {
+            name: instanceName,
+            labels: {
+              workertype: workerType.name,
+            },
+            description: workerType.description,
+            machineType: `zones/${zone}/machineTypes/${workerType.config.machineType}`,
+            scheduling: workerType.config.scheduling,
+            networkInterfaces: workerType.config.networkInterfaces,
+            disks: workerType.config.disks,
+            serviceAccounts: [{
+              email: this.workerAccountEmail,
+              scopes: [
+                /*
+                 * This looks scary but is ok. According to
+                 * https://cloud.google.com/compute/docs/access/service-accounts#accesscopesiam
+                 *
+                 * "A best practice is to set the full cloud-platform
+                 * access scope on the instance, then securely limit
+                 * the service account's API access with IAM roles."
+                 *
+                 * Which is what we do.
+                 */
+                'https://www.googleapis.com/auth/cloud-platform',
+              ],
+            }],
+            metadata: {
+              items: [
+                {
+                  key: 'taskcluster',
+                  value: JSON.stringify({
+                    provisionerId: this.provisionerId,
+                    workerType: workerType.name,
+                    workerGroup: `${workerType.name}-google`,
+                    credentialUrl: libUrls.api(this.rootUrl, 'worker-manager', 'v1', `credentials/google/${workerType.name}`),
+                    rootUrl: this.rootUrl,
+                    userData: workerType.config.userData,
+                  }),
+                },
+              ],
+            },
+          },
+        });
+        op = res.data;
+      } catch (err) {
+        if (!err.errors) {
+          throw err;
+        }
+        for (const error of err.errors) {
+          await workerType.reportError({
+            kind: 'creation-error',
+            title: 'Instance Creation Error',
+            description: error.message, // TODO: Make sure we clear exposing this with security folks
```

Comment:
I never understood the hesitance to show these error messages..

Reply:
Me neither but we were worried before so we can be worried again I guess.

Reply:
I'll include it in the RRA